core/sdk-core/src/main/java/software/amazon/awssdk/core/async/AsyncResponseTransformer.java

```diff
@@ -118,6 +118,26 @@
         return new FileAsyncResponseTransformer<>(path);
     }
 
+    /**
+     * Creates an {@link AsyncResponseTransformer} that writes all the content to the given file.
+     *
+     * @param path            Path to file to write to.
+     * @param position        The value for the data between the current end of the file and the starting position is
+     *                        undefined.
+     * @param isNewFile       Whether this is a new file. If this is {@code true} and the file already exists, the
+     *                        transformer will complete with an exception.
+     * @param deleteOnFailure Whether the file on disk should be deleted in the event of a failure when writing the
+     *                        stream.
+     * @param <ResponseT> Pojo Response type.
+     * @return AsyncResponseTransformer instance.
+     */
+    static <ResponseT> AsyncResponseTransformer<ResponseT, ResponseT> toFile(Path path,
+                                                                             long position,
```

Comment:
I wonder if we should use a request wrapper for this. Something like `toFile(Path path, FileOption, option)`? My concern is that we might need to add other parameters in the future.

Reply:
I think in that case case a `Builder` might be a better option, something like

```java
AsyncResponseTransformer.toFileTransformerBuilder()
                        .position(...)
                        .deleteOnFailure(...)
                        .build();
```
We have one for `FileAsyncRequestBody` https://github.com/aws/aws-sdk-java-v2/blob/a7519245f57c1ac0e57b79f094a0274c715726ee/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/FileAsyncRequestBody.java#L103

Reply:
Like the idea of builder but I'm not sure if we want to put `toFileTransformerBuilder` api to `AsyncResponseTransformer`.

What would `toFile` look like? `toFile(FileAsyncResponseTransformer.Builder builder)`?

Reply:
We can probably just keep `toFile(Path)` as-is, since it's behavior is probably what most customers want. This extra stuff is for less common/more specialized cases I would think (such as use in the transfer manager).